h5py/_hl/files.py

```diff
@@ -233,6 +237,9 @@ def make_fid(name, mode, userblock_size, fapl, fcpl=None, swmr=False):
         # Not all drivers raise FileNotFoundError (commented those that do not)
         except FileNotFoundError if fapl.get_driver() in (
             h5fd.SEC2,
+            # 2022/01/19 hmaarrfk
+            # TODO: I'm not sure if this actually raises FileNotFoundError
+            h5fd.DIRECT if direct_vfd else -1,
```

Comment:
Can you check? It should just be a matter of opening a nonexistant file with this driver in mode `'r'` or `'r+'` to see what error you get.

Reply:
```
import h5py
h5py.File("sdkfjsl", "r", driver="direct")

FileNotFoundError: [Errno 2] Unable to open file (unable to open file, errno = 2, error message = 'No such file or directory')
```
Seems to confirm that this is the correct error to catch. Thank you!